### PR TITLE
Add new flag to authors tool 

### PR DESCRIPTION
**Pull request by rlucas7**
#### Reference issue
https://mail.python.org/pipermail/scipy-dev/2020-August/024331.html

#### What does this implement/fix?
Adds a `--new` or `-n` short flag to the `tools/authors.py` module. 

#### Additional information
Using the flag added in this PR will autogenerate the list of new authors to add to the scipy.org website. 
We'd still need to copy and paste the lines into the file 
https://github.com/rlucas7/scipy.org/blob/master/www/scipylib/donations.rst
and open a PR on that repo but otherwise that would be the extent of work effort. 

The `--new` flag does not provide what the contributor contributed to the scipy repo
so it may make sense to remove the additional context on:

https://github.com/scipy/scipy.org/pull/364

so we have a list of names only. 
Open to suggestions here if someone has a better idea. 

#### Example Usage 
Note here that 
```
4c0fd7939  -> 1.5.0 tag 
f533e9103  -> 1.4.0 tag 
```

So the current way to use the tool to get all authors between 1.4.0 and 1.5.0 release is:
```
python tools/authors.py f533e9103..4c0fd7939
```

If just the new authors are wanted for names to add to the scipy.org website repo use: 

```
python tools/authors.py f533e9103..4c0fd7939 --new
```

For me this latter command gives output: 

```
(scipy-dev) Lucass-MacBook:scipy rlucas$ python tools/authors.py f533e9103..4c0fd7939 -n | head -n 15


- Kale-ab Tessera
- Kevin Mader
- poom
- Berkay Antmen
- Milad Sadeghi DM
- Jan Vleeshouwers
- Matthias Bussonnier
- Shubham Mishra
- jeremie du boisberranger
- Jordão Bragantini
- Arne Küderle
- Nicholas McKibben
- James Wright
```

### CODE REVIEW

**Comment by rlucas7** [comment]
I didn't realize I was on the remove thanks branch so that crept in on the push that created this PR. 

I did a rebase and dropped the commit that removed the thanks file so this PR is separate from the other change.

**Comment by rgommers** [comment]
> I did a rebase and dropped the commit that removed the thanks file so this PR is separate from the other change.

I think you removed the wrong commit, current it contains only the `THANKS.txt` deletion.

**Comment by rlucas7** [comment]
> > I did a rebase and dropped the commit that removed the thanks file so this PR is separate from the other change.
> 
> I think you removed the wrong commit, current it contains only the `THANKS.txt` deletion.

Ugh, yeah
```
git fsck --lost-found
```
got it back for me and removes Thanks.txt file remove commit.

File: tools/authors.py [hunk 1]
```diff
@@ -97,6 +99,16 @@ def name_key(fullname):
             surname = surname[4:]
         return (surname.lower(), forename.lower())
 
+    # generate set of all new authors
+    if vars(options)['new']:
+        new_authors = authors.difference(all_authors)
+        # Print some empty lines to separate
+        stdout_b.write(("\n\n").encode('utf-8'))
```
Comment (rgommers) [inline tools/authors.py:106]: Could you also add the alphabetic sorting here? I.e. copy lines 112-113.

**Review by rgommers — COMMENTED** [review]
Thanks @rlucas7, this LGTM, only one small request

File: tools/authors.py (hunk 1 above)
Comment (rlucas7) [reply tools/authors.py:108]: > Could you also add the alphabetic sorting here? I.e. copy lines 112-113.

yep, in latest changes pushed.

**Review by rgommers — APPROVED** [review]
LGTM now, merging. Thanks @rlucas7!

**Comment by rlucas7** [comment]
Thanks Ralf. 

Do we want to do:

https://github.com/scipy/scipy/pull/12792

now too?

Or do we want to wait for the next release?

I was thinking removal of those now might reduce conflicts during (more) hectic times closer to release. 
 
-Lucas

**Pull request merged**